### PR TITLE
[KYUUBI #2021] Command OptionParser for launching Hive Backend Engine

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
how about simply

```scala
def fromCommandLineArgs(args: Seq[String], conf: KyuubiConf): Unit = {
    for (i <- args.indices by 2) {
      require(args(i) == "--conf", s"Unrecognized main arguments prefix ${args(i)}")
      require(i + 1 <= args.size , "Command arguments ended unexpectedly")

      args(i + 1).split("=", 2) match {
        case seq if seq.size == 2 => conf.set(seq.head, seq.last)
        case _ => throw new IllegalArgumentException(s" $args(i + 1)")
      }
    }
  }
```

**Comment by zhaomin1423** [comment]
> how about simply
> 
> ```scala
> def fromCommandLineArgs(args: Seq[String], conf: KyuubiConf): Unit = {
>     for (i <- args.indices by 2) {
>       require(args(i) == "--conf", s"Unrecognized main arguments prefix ${args(i)}")
>       require(i + 1 <= args.size , "Command arguments ended unexpectedly")
> 
>       args(i + 1).split("=", 2) match {
>         case seq if seq.size == 2 => conf.set(seq.head, seq.last)
>         case _ => throw new IllegalArgumentException(s" $args(i + 1)")
>       }
>     }
>   }
> ```

but, as if it's inconvenient if need some other arguments.

**Comment by yaooqinn** [comment]
> > how about simply
> > ```scala
> > def fromCommandLineArgs(args: Seq[String], conf: KyuubiConf): Unit = {
> >     for (i <- args.indices by 2) {
> >       require(args(i) == "--conf", s"Unrecognized main arguments prefix ${args(i)}")
> >       require(i + 1 <= args.size , "Command arguments ended unexpectedly")
> > 
> >       args(i + 1).split("=", 2) match {
> >         case seq if seq.size == 2 => conf.set(seq.head, seq.last)
> >         case _ => throw new IllegalArgumentException(s" $args(i + 1)")
> >       }
> >     }
> >   }
> > ```
> 
> but, as if it's inconvenient if need some other arguments.

any cases

**Comment by zhaomin1423** [comment]
> > > how about simply
> > > ```scala
> > > def fromCommandLineArgs(args: Seq[String], conf: KyuubiConf): Unit = {
> > >     for (i <- args.indices by 2) {
> > >       require(args(i) == "--conf", s"Unrecognized main arguments prefix ${args(i)}")
> > >       require(i + 1 <= args.size , "Command arguments ended unexpectedly")
> > > 
> > >       args(i + 1).split("=", 2) match {
> > >         case seq if seq.size == 2 => conf.set(seq.head, seq.last)
> > >         case _ => throw new IllegalArgumentException(s" $args(i + 1)")
> > >       }
> > >     }
> > >   }
> > > ```
> > 
> > 
> > but, as if it's inconvenient if need some other arguments.
> 
> any cases

None, all can be replaced by conf.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+package org.apache.kyuubi.util
+
+import org.apache.kyuubi.config.KyuubiConf
+
+object ArgsParser {
+
+  def fromCommandLineArgs(args: Array[String], conf: KyuubiConf): Unit = {
+    for (i <- args.indices by 2) {
+      require(args(i) == "--conf", s"Unrecognized main arguments prefix ${args(i)}")
+      require(i + 1 <= args.length, "Command arguments ended unexpectedly")
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala:26]: maybe we can check this once

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.util
+
+import org.apache.kyuubi.config.KyuubiConf
+
+object ArgsParser {
+
+  def fromCommandLineArgs(args: Array[String], conf: KyuubiConf): Unit = {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala:23]: put it in Utils or rename the class
 to Command Line Utils?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala (hunk 1 above)
Comment (zhaomin1423) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala]: done

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala (hunk 2 above)
Comment (zhaomin1423) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala]: move it to Utils

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala (hunk 1 above)
Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala]: No in this way, I mean checking out of the loop

Comment (zhaomin1423) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ArgsParser.scala]: done

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**